src/features/vault.ts

```diff
@@ -0,0 +1,247 @@
+import * as vscode from 'vscode';
+import * as utilAnsibleCfg from './utils/ansibleCfg';
+import * as tmp from 'tmp';
+import * as cp from 'child_process';
+import * as fs from 'fs';
+
+async function askForVaultId(ansibleCfg: string) {
+  let vaultId = 'default';
+  let identityList: string[] | undefined;
+  if (ansibleCfg === 'ANSIBLE_VAULT_IDENTITY_LIST') {
+    const envVar: string | undefined = process.env.ANSIBLE_VAULT_IDENTITY_LIST;
+    if (envVar !== undefined) {
+      identityList = envVar
+        .split(',')
+        .map((id: string) => id.split('@', 2)[0].trim());
+    } else {
+      identityList = [];
+    }
+  } else {
+    const cfg: utilAnsibleCfg.AnsibleVaultConfig | undefined =
+      utilAnsibleCfg.getValueByCfg(ansibleCfg);
+    if (cfg !== undefined) {
+      identityList = cfg.defaults.vault_identity_list
+        .split(',')
+        .map((id: string) => id.split('@', 2)[0].trim());
+    } else {
+      identityList = [];
+    }
+  }
+  if (identityList.length !== 0) {
+    await vscode.window.showQuickPick(identityList).then((chosen) => {
+      if (chosen) {
+        vaultId = chosen;
+      }
+    });
+    return vaultId;
+  } else {
+    return undefined;
+  }
+}
+
+function displayMissingIdentityError(): void {
+  vscode.window.showErrorMessage(
+    'no ansible vault identity defined, cannot de-/encrypt'
+  );
+}
+
+export const toggleEncrypt = async (): Promise<void> => {
+  const editor = vscode.window.activeTextEditor;
+  if (!editor) {
+    return;
+  }
+
+  const selection = editor.selection;
+  if (!selection) {
+    return;
+  }
+
+  const config = vscode.workspace.getConfiguration('ansible.vault');
+  const doc = editor.document;
+
+  // Read `ansible.cfg` or environment variable
+  const rootPath: string | undefined = utilAnsibleCfg.getRootPath(
+    editor.document.uri
+  );
+  let keyInCfg: string;
+  if (!!process.env.ANSIBLE_VAULT_IDENTITY_LIST) {
+    keyInCfg = 'ANSIBLE_VAULT_IDENTITY_LIST';
+  } else {
+    keyInCfg = utilAnsibleCfg.scanAnsibleCfg(rootPath);
+  }
```

Comment:
This should really be a `const`
```suggestion
  const keyInCfg: string = !!process.env.ANSIBLE_VAULT_IDENTITY_LIST
    ? 'ANSIBLE_VAULT_IDENTITY_LIST'
    : utilAnsibleCfg.scanAnsibleCfg(rootPath);
```